src/coreclr/src/System.Private.CoreLib/src/System/CLRConfig.cs

```diff
@@ -28,7 +28,7 @@ internal static bool GetBoolValueWithFallbacks(string switchName, string environ
             // abstractions where reasonably possible.
 
             Span<char> buffer = stackalloc char[32];
-            int length = Interop.Kernel32.GetEnvironmentVariable(environmentName, buffer);
+            uint length = Interop.Kernel32.GetEnvironmentVariable(environmentName, ref buffer.GetPinnableReference(), (uint)buffer.Length);
```

Comment:
Out of interest, why did you use `GetPinnableReference` and not `MemoryMarshal.GetReference` as we know `buffer`'s length is > 0, so no null-ref will occur?

IMHO the code as you wrote is more readable (with the instance method), and perf-wise there won't be any measurable difference.

Reply:
As you have said, it is more readable and there won't be any measurable difference. 

The JIT will actually optimize the length check away and so the generated code will be identical between the options: https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAJI8YUCAAcAymIBuvMDFwBuGsQDMTUgwDCDAN40GxhgG0AIgBtLQ/BOgYAFACIA1mK4xL60iwAm1s4AlAC6RiYSULyy2BgwTORIDDAIcVBcDBy8PAwA4jAYAKJc8uJcAjwAathR2MCWMI6wAGYMYAAWNQwSaJnZGAwNXADmGO1BygzhxtNMmsQoDACyjkEGsyYMUhLYXAA8HTUAfAzAHM3NYgwAvAy4GNhgrtjWEGBtnVCmPiGq1JubfJFEq8MoVDDVWr1RotU7nS5QFhAgAK2S4dQabBgCJgXEUq16jiyPCCZwuYhYABlcaNxn9NgBfGgMoA==